python/pyspark/worker.py

```diff
@@ -268,9 +272,11 @@ def main(infile, outfile):
 
         # set up memory limits
         memory_limit_mb = int(os.environ.get('PYSPARK_EXECUTOR_MEMORY_MB', "-1"))
-        total_memory = resource.RLIMIT_AS
-        try:
-            if memory_limit_mb > 0:
+        # 'PYSPARK_EXECUTOR_MEMORY_MB' should be undefined on Windows because it depends on
+        # resource package which is a Unix specific package.
+        if memory_limit_mb > 0:
```

Comment:
It seems brittle to disable this on the JVM side and rely on it here. Can we also set a flag in the ImportError case and also check that here?